keras_nlp/models/gpt_neo_x/gpt_neo_x_backbone.py

```diff
@@ -0,0 +1,160 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+from tensorflow import keras
+
+from keras_nlp.api_export import keras_nlp_export
+from keras_nlp.models.backbone import Backbone
+from keras_nlp.models.gpt_neo_x.gpt_neo_x_decoder import GPTNeoXDecoder
+
+
+def _gpt_neo_x_kernel_initializer(stddev=0.02):
+    return keras.initializers.RandomNormal(stddev=stddev)
+
+
+@keras_nlp_export("keras_nlp.models.GPTNeoXBackbone")
+class GPTNeoXBackbone(Backbone):
+    """GPT-2 core network with hyperparameters.
+
+    This network implements a Transformer-based decoder network,
+    Generative Pretrained Transformer-Neo-X (GPTNeoX), as described in
+    ["GPT-NeoX-20B: An Open-Source Autoregressive Language Model"](https://arxiv.org/abs/2204.06745).
+    It includes the embedding lookups and transformer layers.
+
+    The default constructor gives a fully customizable, randomly initialized
+    GPT-NeoX model with any number of layers, heads, and embedding
+    dimensions.
+
+    Disclaimer: Pre-trained models are provided on an "as is" basis, without
+    warranties or conditions of any kind. The underlying model is provided by a
+    third party and subject to a separate license, available
+    [here](https://github.com/EleutherAI/pythiarota).
```

Comment:
I think we should point this to the original codebase for gpt-neox right? https://github.com/EleutherAI/gpt-neox/